src/machine/machine_stm32_moder_gpio.go

```diff
@@ -0,0 +1,143 @@
+// +build stm32,!stm32f103xx
+
+package machine
+
+// GPIO for the stm32 families except the stm32f1xx which uses a simpler but
+//  less flexible mechanism. Extend the +build directive above to exclude other
+//  models in the stm32f1xx series as necessary
+
+const (
+	// Mode Flag
+	PinOutput        PinMode = 0
+	PinInput         PinMode = PinInputFloating
+	PinInputFloating PinMode = 1
+	PinInputPulldown PinMode = 2
+	PinInputPullup   PinMode = 3
+
+	// for UART
+	PinModeUARTTX PinMode = 4
+	PinModeUARTRX PinMode = 5
+
+	// for I2C
+	PinModeI2CSCL PinMode = 6
+	PinModeI2CSDA PinMode = 7
+
+	// for SPI
+	PinModeSPICLK  PinMode = 8
+	PinModeSPIMOSI PinMode = 9
+	PinModeSPIMISO PinMode = 10
+
+	// for analog/ADC
+	PinInputAnalog PinMode = 11
+
+	// for PWM
+
+	// Register values for the chip
+	// GPIOx_MODER
+	GPIOModeInput         = 0
```

Comment:
I guess you added these bitfields here because they don't exist in the stm32 package?

It might be better to move them over to the stm32 package (like src/device/stm32/stm32f103xx-bitfields.go) or maybe switch to the (hopefully more complete) stm32-rs package. But that can also be done at a later time.